framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -592,195 +497,285 @@ public void standardizeVariable(AnnotatedTypeMirror type, Element elt) {
-                        receiver,
-                        JavaExpression.getParametersOfEnclosingMethod(localVarPath),
-                        factory.getChecker());
-        standardizeAtm(localContext, localVarPath, type, /*removeErroneousExpressions=*/ false);
+    public void delocalize(MethodTree methodDeclTree, AnnotatedTypeMirror atm) {
+        if (!hasDependentType(atm)) {
+            return;
+        }
+
+        TreePath pathToMethodDecl = factory.getPath(methodDeclTree);
+
+        ExecutableElement methodElement = TreeUtils.elementFromDeclaration(methodDeclTree);
+        StringToJavaExpression stringToJavaExpr =
+                expression -> {
+                    JavaExpression result;
+                    try {
+                        result =
+                                StringToJavaExpression.atPath(
+                                        expression, pathToMethodDecl, factory.getChecker());
+                    } catch (JavaExpressionParseException ex) {
+                        return null;
+                    }
+                    List<FormalParameter> parameters =
+                            JavaExpression.getFormalParameters(methodElement);
+                    List<JavaExpression> paramsAsLocals =
+                            JavaExpression.getParametersAsLocalVariables(methodElement);
+                    JavaExpressionConverter jec =
+                            new JavaExpressionConverter() {
+                                @Override
+                                protected JavaExpression visitLocalVariable(
+                                        LocalVariable localVarExpr, Void unused) {
+                                    int index = paramsAsLocals.indexOf(localVarExpr);
+                                    if (index == -1) {
+                                        return new Unknown(
+                                                localVarExpr.getType(),
+                                                localVarExpr
+                                                        .getElement()
+                                                        .getSimpleName()
+                                                        .toString());
+                                    }
+                                    return parameters.get(index);
+                                }
+                            };
+                    result = jec.convert(result);
+                    return result.containsUnknown() ? null : result;
+                };
+        convertAnnotatedTypeMirror(stringToJavaExpr, atm);
     }
 
     /**
-     * Viewpont-adapt the dependent types in {@code type} using the {@code context} provided.
+     * Calls {@link #convertAnnotationMirror(StringToJavaExpression, AnnotationMirror)} on each
+     * annotation mirror on type with {@code stringToJavaExpr}. And replaces the annotation with the
+     * one created by {@code convertAnnotationMirror}, it it's not null. See {@link
```

Comment:
"it it's" => "if it's".

Please document what happens if it *is* null.  I think the value is dropped, though on my reading of this Javadoc it led me to believe that the original value would be retained unchanged.